### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by kong75**

### CODE REVIEW

File: back-end/src/api/task_api_v1.py [hunk 1]
```diff
@@ -155,6 +156,23 @@ def google_search_func():
...
+        data = json.loads(request.data)
+        video_url = data["video_url"]
+        input_variables = data.get("input_variables", None)
+
+        res = youtube_transcript(video_url, input_variables=input_variables)
+
+        return jsonify({"result": res})
+
+    except ValueError as e:
+        return jsonify({"error": str(e)}), 400
```
Comment (Spaceenter) [inline back-end/src/api/task_api_v1.py:173]: Usually we log the detailed error message, but return a general error.

File: back-end/src/core/component/youtube_transcript.py [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+from urllib.error import HTTPError, URLError
+from config import logger
+import os
+
+class YouTubeTranscript:
+    """YouTube Transcript class."""
+
+    def __init__(self):
+        pass
+
```
Comment (Spaceenter) [inline back-end/src/core/component/youtube_transcript.py:12]: You can delete __init__()

**Review by Spaceenter — APPROVED** [review]

File: back-end/src/api/task_api_v1.py (hunk 1 above)
Comment (kong75) [reply back-end/src/api/task_api_v1.py:173]: Currently this is aligned with other functions and we may refactor together in the future.

File: back-end/src/core/component/youtube_transcript.py (hunk 2 above)
Comment (kong75) [reply back-end/src/core/component/youtube_transcript.py:12]: Fixed, thank you!

**Pull request merged**